Bottom: 064b9d0a4da815235672b3effa77065f13d8b467
Top:    0b92297b8ae44565cf3a2b253899ea73ebed8c13
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2010-09-09 16:16:29 +0100

Temp fixes and diagnostics


---

diff --git a/src/polygon1.c b/src/polygon1.c
index 6600667..caf8807 100644
--- a/src/polygon1.c
+++ b/src/polygon1.c
@@ -1163,7 +1163,11 @@ label_contour (PLINE * a)
       assert (label == INSIDE || label == OUTSIDE);
       LABEL_NODE (cur, label);
     }
-  while ((cur = cur->next) != first_labelled);
+  while ((cur = cur->next) != first_labelled &&
+	 !(cur->next == &a->head && first_labelled == NULL));
+
+  if (first_labelled == NULL)
+    fprintf (stderr, "OOPS, DIDN'T LABEL ANYTHING!\n");
 #ifdef DEBUG_ALL_LABELS
   print_labels (a);
   DEBUGP ("\n\n");
diff --git a/src/sweep.h b/src/sweep.h
new file mode 100644
index 0000000..e69de29
